src/mono/wasm/runtime/startup.ts

```diff
@@ -316,7 +312,7 @@ export async function mono_load_runtime_and_bcl_args(args: MonoConfig): Promise<
 
         _apply_configuration_from_args(args);
 
-        const local_fetch = _get_fetch_implementation(args);
+        const local_fetch = typeof (args.fetch_file_cb) === "function" ? args.fetch_file_cb : _fetch_asset;
```

Comment:
The fact that the extension point function is on config file which is otherwise loaded from json is bit unfortunate.
I would love to abandon this `fetch_file_cb`, rename it to `fetch` and look for it on `Module.imports.fetch` where `Module.imports` is just container object for all environment things which could be replaced.

Reply:
After modularization `imports` would live on `EmscriptenModuleConfig` type

Reply:
```
    createDotnetRuntime(({ MONO, INTERNAL, BINDING, Module }) => ({
        imports: {
            fetch: () => { return something(); },
            ws: { WebSocket:MyWebSocket, somthinngElse },
        },
```

Reply:
yeah I think that would be an imrovement.

Reply:
yeah, on board with the imports thing. another stopgap option would be to change the config fetch_file_cb to be the name of a function that the runtime can find in globalThis, i.e. ```globalThis[config.fetch_file_cb_name]```

Reply:
We could even be very specific about full list of `imports` which could be replaced in our typescript definition.

Reply:
Okay, I could actually do that in some later PR.